com/win32com/client/__init__.py

```diff
@@ -315,11 +315,13 @@ class object that derives from three classes:
             # Get the class from the module.
             disp_class = gencache.GetClassForProgID(str(disp_clsid))
         except pythoncom.com_error:
-            raise TypeError(
-                "This COM object can not automate the makepy process - please run makepy manually for this object"
-            )
+            pass
     else:
         disp_class = disp.__class__
+    if disp_class is None:
```

Comment:
thanks! This could all still stay in that inner `if` block

Reply:
Technically yes, but neither mypy nor pyright seem to see that `disp_class` cannot be `None` if I raise after the try-except in the `if` branch.
Some likely relevant issues:
- https://github.com/python/mypy/issues/14309
- https://github.com/python/mypy/issues/12782
- https://github.com/python/mypy/issues/14527

Reply:
Deduplicating code between DispatchWithEvents and WithEvents resolved this anyway